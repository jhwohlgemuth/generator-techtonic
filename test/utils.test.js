'use strict';

const sinon        = require('sinon');
const {last}       = require('lodash');
const {join}       = require('path');
const {unlinkSync} = require('fs');
const {
    fin,
    fail,
    download,
    formatFederalAgencyData
} = require('../generators/server/data-utils');
const {
    json,
    maybeInclude,
    parseModuleData
} = require('../generators/app/utils');

const ORIGINAL_CONSOLE_LOG = window.console.log;
const TEST_JSON_DATA = require('./data/test.json');

describe('Data Utilities Module', () => {
    beforeAll(() => {
        window.console.log = jest.fn().mockName('Console Log');
    });
    afterAll(() => {
        window.console.log.mock.mockReset();
        window.console.log = ORIGINAL_CONSOLE_LOG;
    });
    it('can log failure message for invalid url', () => {
        const url = 'not a valid url';
        const path = url;
        download({url, path});
        expect(window.console.log.mock.calls).toMatchSnapshot();
    });
    it('can format federal agency JSON data', () => {
        const results = formatFederalAgencyData(TEST_JSON_DATA);
        expect(results).toMatchSnapshot();
    });
    it('can log success and fail messages via promise chain', () => {
        const path = 'some-path-name.json'
        expect(fin(path)([1, 2, 3])).toMatchSnapshot();
        expect(fail(path)([4, 5, 6])).toMatchSnapshot();
        expect(window.console.log.mock.calls).toMatchSnapshot();
    });
});
describe('Utilities Module', () => {
    it('can parse module data string', () => {
        expect(
            [
                'AMD with \u001b[31mr.js\u001b[39m',
                'AMD with \u001b[33mbrowserify\u001b[39m',
                'CJS with \u001b[33mbrowserify\u001b[39m',
                'CJS with \u001b[34mwebpack\u001b[39m'
            ].map(parseModuleData).map(last)
        ).toEqual([
            'rjs',
            'browserify',
            'browserify',
            'webpack'
        ]);
    });
    it('can optionally return values', () => {
        const result = [].concat(
            maybeInclude(true, 'foo', 'bar'),
            maybeInclude(false, 'a', 'b'),
            maybeInclude(true, ['c', 'd'])
        );
        expect(result).toMatchSnapshot();
    });
});
describe('Generator JSON utility', () => {
    let content;
    const testJSON = join(__dirname, 'test.json');
    beforeEach(() => {
        json.write(testJSON, {foo: 'bar'});
    });
    afterEach(() => {
        unlinkSync(testJSON);
    });
    it('can extend JSON files', () => {
        content = json.read(testJSON);
        expect(content).toMatchSnapshot();
        json.extend(testJSON, {bar: 'baz'});
        content = json.read(testJSON);
        expect(content).toMatchSnapshot();
    });
});
