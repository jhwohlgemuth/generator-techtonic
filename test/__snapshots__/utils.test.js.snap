// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Utilities Module can format federal agency JSON data 1`] = `
Array [
  Object {
    "federalAgency": "Department of Commerce",
    "isCfo": "Y",
    "subAgency": "National Institute of Standards and Technology",
    "term": "nist-gov",
    "uniqueId": "nist-gov",
  },
  Object {
    "federalAgency": "Department of Justice",
    "isCfo": "Y",
    "subAgency": "Office of Legal Counsel",
    "term": "olc-usdoj-gov",
    "uniqueId": "olc-usdoj-gov",
  },
  Object {
    "federalAgency": "Department of State",
    "isCfo": "Y",
    "subAgency": "",
    "term": "state-gov",
    "uniqueId": "state-gov",
  },
]
`;

exports[`Data Utilities Module can log failure message for invalid url 1`] = `Array []`;

exports[`Data Utilities Module can log success and fail messages via promise chain 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`Data Utilities Module can log success and fail messages via promise chain 2`] = `
Array [
  4,
  5,
  6,
]
`;

exports[`Data Utilities Module can log success and fail messages via promise chain 3`] = `
Array [
  Array [
    "[31m[1mâœ—[22m[39m Failed to download [36mnot a valid url[39m",
  ],
  Array [
    "[32m[1mâœ”[22m[39m Successfully downloaded [36msome-path-name.json[39m",
  ],
  Array [
    "[31m[1mâœ—[22m[39m Failed to download [36msome-path-name.json[39m",
  ],
]
`;

exports[`Generator JSON utility can extend JSON files 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`Generator JSON utility can extend JSON files 2`] = `
Object {
  "bar": "baz",
  "foo": "bar",
}
`;

exports[`Utilities Module can optionally return values 1`] = `
Array [
  "foo",
  "b",
  "c",
  "d",
]
`;
